From 5eef88f122e918971660ff3ebdc4b815acab40d6 Mon Sep 17 00:00:00 2001
From: Christoph Paasch <cpaasch@openai.com>
Date: Fri, 1 Aug 2025 14:00:28 -0700
Subject: [PATCH 05/17] [nh] Add missing free() in rtnl_nh_grp_put

nhg->entries is dynamically allocated and needs to be free'd.

This has been there since commit 780d06ae8bee ("route: add nh type")

Fixes: 780d06ae8bee ("route: add nh type")
Signed-off-by: Christoph Paasch <cpaasch@openai.com>
Upstream-Status: Backport [https://github.com/thom311/libnl/commit/7c27396423dcaf4a5ebf77cf9df9234923bbe115]
Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 lib/route/nh.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/route/nh.c b/lib/route/nh.c
index 3dfe5586dcac..b2fb9db52f15 100644
--- a/lib/route/nh.c
+++ b/lib/route/nh.c
@@ -85,6 +85,7 @@ static void rtnl_nh_grp_put(nl_nh_group_t *nhg)
 	if (nhg->ce_refcnt > 0)
 		return;
 
+	free(nhg->entries);
 	free(nhg);
 }
 
-- 
2.51.0

